core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -268,8 +270,7 @@ CloseableIterable<ManifestEntry<F>> liveEntries() {
   /** @return an Iterator of DataFile. Makes defensive copies of files before returning */
   @Override
   public CloseableIterator<F> iterator() {
-    return CloseableIterable.transform(liveEntries(), e -> e.file().copy(!dropStats(columns)))
-        .iterator();
+    return CloseableIterable.transform(liveEntries(), e -> e.file().copy(!dropStats)).iterator();
```

Comment:
I think this is where the value of `dropStats` should be set, as a local variable. Otherwise, when the reader is configured, the value of `dropStats` gets stale. For example, when `project` is called.

Reply:
good catch, I should have definitely checked the surrounding code and not just the diff on the PR. Given that tests also didn't catch that, I think it would be a good idea to try and add a test for this

Reply:
@rdblue @nastra I missed that `dropStats` is invoked at internal lambda function. It's fixed and I also try adding a UT for project. However, with project `columns == null` is always true so `dropStats` is always false. I doubt whether it's a meaningful test.